src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+//ProgressBar
+interface ProgressBarProps {
+  duration: number;
+  type?: "linear" | "circular";
+}
+   
+const ProgressBar: React.FC<ProgressBarProps> = ({ duration, type = "linear" }) => {
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
```

Comment:
The interval increases progress until it reaches 100, but the interval keeps running even after reaching 100. A possible improvement would be to clear the interval once progress reaches 100 to avoid unnecessary state updates and computations.